docs/design/INTEGRATE-HEALTHCHECK-SERVICE.md

```diff
@@ -0,0 +1,62 @@
+# Integrating EdFi.AdminConsole.HealthCheckService into EdFi.Ods.AdminApi with Quartz.NET
+
+## Overview
+
+# Integrating EdFi.Ods.AdminApi.HealthCheck into EdFi.Ods.AdminApi with Quartz.NET
+
+## Overview
+
+This document describes the design and process for integrating the `EdFi.Ods.AdminApi.HealthCheck` into the `EdFi.Ods.AdminApi` application, leveraging Quartz.NET for scheduled and on-demand execution of health checks.
+
+---
+
+## Goals
+
+* Enable scheduled health checks of ODS API instances via Quartz.NET.
+* Allow on-demand triggering of health checks via an API endpoint.
+* Ensure only one health check job runs at a time to prevent data conflicts.
+* Centralize health check logic in `EdFi.Ods.AdminApi.HealthCheck`.
+
+---
+
+## Architecture
+
+### Components
+
+* **HealthCheckService**: Service class that performs health checks across tenants and instances.
+* **HealthCheckJob**: Quartz.NET job that invokes `HealthCheckService.Run()`.
+* **Quartz.NET Scheduler**: Manages scheduled and ad-hoc job execution.
+* **HealthCheckTrigger Endpoint**: API endpoint to trigger health checks on demand.
+
+---
+
+## Process Flow
+
+### 1. Service Registration
+
+* Register `HealthCheckService` and its dependencies in the DI container (typically as `scoped` or `transient`).
+* Register `HealthCheckJob` with Quartz.NET using `AddQuartz` and `AddQuartzHostedService`.
+
+### 2. Scheduling with Quartz.NET
+
+* Configure Quartz.NET to schedule `HealthCheckJob` at a configurable interval (e.g., every 10 minutes, using `HealthCheckFrequencyInMinutes` from configuration).
+* Use the `[DisallowConcurrentExecution]` attribute on `HealthCheckJob` to prevent overlapping executions.
+
+### 3. On-Demand Triggering
+
+* Implement an API endpoint (e.g., `/healthcheck/trigger`) in `EdFi.Ods.AdminApi`. Note: Grouped with `adminconsole` endpoints for consistency.
```

Comment:
1. Let's put this in `/v2` instead of `/adminconsole`.
2. Can we simply use `POST /adminconsole/healthcheck` instead of adding "trigger"?
3. What is the payload for this endpoint?

Reply:
I think we need a payload, or we need a modification to the API endpoint, because we shouldn't be triggering a health check on _every_ ODS instance. A triggered health check should be on a specific instance.